### DISCUSSION TITLE
Any ideas why this component does not respond to changes to it's attribute?

### DISCUSSION THREAD

**Discussion by TheMouster** [defx/synergy#15] [General]
```
import { define } from '/lib/synergy/dist/synergy.min.js';

const DEFAULT = '-';

const numericFormatter = (places) => new Intl.NumberFormat(navigator.language, { minimumFractionDigits: places, maximumFractionDigits: places });

const currencyValueFactory = (currencyElement) => {
    let money = parseFloat(currencyElement.getAttribute("money"));

    if (isNaN(money)) return { DEFAULT };

    let formatted = money !== 0.0 ? numericFormatter(2).format(money) : DEFAULT;

    return { formatted };
};

define('currency-value', currencyValueFactory, "{{ formatted }}");
```

it renders perfectly at load, but if I update the currency-value custom element's "money" attribute the change is not updated. If I inspect the page I can see the currency value being updated from it's proxy but nothing shows on the front end and the code that renders the component is not executed.

TIA

**Comment by defx** [comment]
Hi @TheMouster, work on this project stalled around 5-6 months ago. I had a big re-write in the works but have since dropped the idea of reactive templating altogether in favour of a different approach. I will update the repo to reflect the fact that Synergy is no longer maintained.

**Comment by TheMouster** [comment]
Pity, it actually works reasonably well, bar the reactive bit :-). What is your different approach, If you don't mind me asking?
As it happens I'm under a bit of self-imposed time-pressure and I've started using Lit which works quite well, so I'm not stuck up a tree so to speak at this point. If I finally get some time in the future and feel like torturing myself with JavaScript I might have a go at making it reactive. Presuming the web hasn't moved on too far from here by then. Thanks for the effort BTW. Synergy has been my go-to tool for a number of component type items including value formatters, which it works wonderfully well for. Only the reactivity was letting me down.
